crates/aptos/src/local_testnet/mod.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::types::{CliCommand, CliError, CliResult, CliTypedResult};
+use aptos_faucet::FaucetArgs;
+use aptos_types::chain_id::ChainId;
+use async_trait::async_trait;
+use clap::Parser;
+use hex::FromHex;
+use rand::{rngs::StdRng, SeedableRng};
+use reqwest::Url;
+use std::{
+    path::PathBuf,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc,
+    },
+    thread,
+    time::{Duration, Instant},
+};
+
+/// CLI Tool for running a local testnet
+///
+#[derive(Parser)]
+pub enum LocalTestnetTool {
+    Node(RunTestNode),
+}
+
+impl LocalTestnetTool {
+    pub async fn execute(self) -> CliResult {
+        use LocalTestnetTool::*;
+        match self {
+            Node(tool) => tool.execute_serialized().await,
+        }
+    }
+}
+
+/// Run a single validator node testnet locally
+#[derive(Parser)]
+pub struct RunTestNode {
+    /// An overridable config for the test node
+    #[clap(long, parse(from_os_str))]
+    config_path: Option<PathBuf>,
+    /// The directory to save all files for the node
+    #[clap(long, parse(from_os_str), default_value = ".aptos/local-testnet")]
+    node_dir: PathBuf,
+    /// Random seed for key generation in test mode
+    #[clap(long, parse(try_from_str = FromHex::from_hex))]
+    seed: Option<[u8; 32]>,
+    /// Run a faucet alongside the node
+    #[clap(long)]
+    with_faucet: bool,
+}
+
+#[async_trait]
+impl CliCommand<()> for RunTestNode {
+    fn command_name(&self) -> &'static str {
+        "RunTestNode"
+    }
+
+    async fn execute(mut self) -> CliTypedResult<()> {
+        let rng = self
+            .seed
+            .map(StdRng::from_seed)
+            .unwrap_or_else(StdRng::from_entropy);
+        let node_dir = self.node_dir.clone();
+        let config_path = self.config_path.clone();
+        // Spawn the node in a separate thread
+        let _node = thread::spawn(move || {
+            aptos_node::load_test_environment(
+                config_path,
+                node_dir,
+                false,
+                false,
+                cached_framework_packages::module_blobs().to_vec(),
+                rng,
+            )
+        });
+
+        // Run faucet if selected
+        let _maybe_faucet = if self.with_faucet {
+            // TODO: Handle if this port is changed
```

Comment:
Yeah this is a bit suss, maybe we should include an argument for the port now?

Reply:
Oh, wait this wasn't supposed to be in this PR...